cli/src/core/client/client.ts

```diff
@@ -20,7 +22,9 @@ export const CreateClient = (opts: ClientOptions): Client => {
 
     // You have to tell the Node.js http API which HTTP version to use.
     httpVersion: '1.1',
-
+    nodeOptions: {
+      agent: opts.proxyUrl ? new HttpsProxyAgent(opts.proxyUrl) : undefined,
```

Comment:
Is it OK that this will propagate an agent property no matter what (even if it's undefined)?

If you don't want to propagate the agent, you might consider ```...(opts.proxyUrl ? { agent: new HttpsProxyAgent(opts.proxyUrl) : {})```

Sometimes this matters.